ssz-rs-derive/src/lib.rs

```diff
@@ -224,10 +225,17 @@ fn derive_deserialize_impl(data: &Data) -> TokenStream {
                     offsets.push((dummy_index, encoding.len()));
 
                     for span in offsets.windows(2) {
+                        // indexes are safe because span is a pair; qed
                         let (index, start) = span[0];
                         let (_, end) = span[1];
 
-                        container.__ssz_rs_set_by_index(index, &encoding[start..end])?;
+                        let target = encoding.get(start..end).ok_or(
```

Comment:
did this in #54 just to avoid the extra computation unless needed
```suggestion
                        let target = encoding.get(start..end).ok_or_else(||
```